File: C++/Problem007.cpp
#include <iostream>
#include <vector>


int main() {
	// Parameters:
	int pos {10001};
	///////////////

	std::vector<int> primes {};

	primes.push_back(2);
	int i {3};

	while(primes.size() < pos) {
		for(auto p : primes) {
			if(p * p > i) {
				primes.push_back(i);
				break;
			} else if (i % p == 0) {
				break;
			}
		}

		i += 2;
	}

	std::cout << primes[pos - 1] << std::endl;

	return 0;
}

File: Julia/Problem007.jl
# Parameters:
pos = 10001
################

primes = [2]
i = 3

while length(primes) < pos
	for p ∈ primes
		if p^2 > i
			append!(primes, i)
			break
		elseif i % p == 0
			break
		end
	end
	
	global i += 2
end

println(primes[end])